commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/arguments/ListArgumentCommon.java

```diff
@@ -137,12 +138,18 @@ public <CommandSourceStack> List<T> parseArgument(CommandContext<CommandSourceSt
 		final String[] strArr = argument.split(Pattern.quote(delimiter));
 		final StringReader context = new StringReader(argument);
 		for (String str : strArr) {
-			if (!values.containsKey(str)) {
+			if (!allowAnyValue && !values.containsKey(str)) {
 				throw new SimpleCommandExceptionType(new LiteralMessage("Item is not allowed in list")).createWithContext(context);
 			} else if (!allowDuplicates && listKeys.contains(str)) {
 				throw new SimpleCommandExceptionType(new LiteralMessage("Duplicate arguments are not allowed")).createWithContext(context);
 			} else {
-				list.add(values.get(str));
+				if (allowAnyValue) {
+					// should only be used with string mappers so we can ignore the cast warning
+					list.add((T) str);
+				}
+				else{
```

Comment:
```suggestion
				else {
```

Reply:
@DerEchtePilz when I use the mapper it will give me a list of nulls. The only way I've found that works is to directly cast it to a string.